calico_node/filesystem/etc/service/felix/run

```diff
@@ -1,5 +1,10 @@
 #!/bin/sh
 exec 2>&1
+# Felix doesn't understand HOSTNAME, but the container exports it as a common
+# interface. This ensures Felix gets the right hostname.
+if [ ! -z $HOSTNAME ]; then
+    export FELIX_FELIXHOSTNAME=$HOSTNAME
```

Comment:
You can remove the if statement and just do the assignment - simplifies the change.